core/src/main/resources/error/error-classes.json

```diff
@@ -630,6 +630,23 @@
       "Input schema <jsonSchema> can only contain STRING as a key type for a MAP."
     ]
   },
+  "INVALID_LIKE_PATTERN" : {
+    "message" : [
+      "The pattern <pattern> is invalid."
+    ],
+    "subClass" : {
+      "ESC_IN_THE_MIDDLE" : {
```

Comment:
sub class order is wrong, should be `ESC_AT_THE_END` then `ESC_IN_THE_MIDDLE`

https://github.com/apache/spark/pull/38658 for fix this